### PR TITLE
Add Simple State Checker document.

### LINKED ISSUE #8895: Add document for SimpleStateChecker & Update State Provider document.
https://github.com/abpframework/abp/pull/8742
https://github.com/abpframework/abp/pull/8663

### PR DESCRIPTION
**Pull request by maliming**
Resolve #8895

### CODE REVIEW

File: docs/en/SimpleStateChecker.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+var myObjectBatchStateChecker = new MyObjectBatchStateChecker();
+
+myobj1.SimpleStateCheckers.Add(myObjectBatchStateChecker);
+myobj2.SimpleStateCheckers.Add(myObjectBatchStateChecker);
+
+SimpleStateCheckerResult<MyObject> stateCheckerResult = await stateCheckerManager.IsEnabledAsync(new []{ myobj1, myobj2 });
+````
+
+## Built-in state checker
```
Comment (hikalkan) [inline docs/en/SimpleStateChecker.md]:
```suggestion
## Built-in State Checkers
```

File: docs/en/SimpleStateChecker.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+services.Configure<AbpSimpleStateCheckerOptions<MyObject>>(options =>
+{
+    options.GlobalSimpleStateCheckers.Add<MyGlobalObjectStateChecker>();
+    //options.GlobalSimpleStateCheckers.Add<>(); //Add more global state checkers
+});
+````
+
+> Write this inside the `ConfigureServices` method of your module.
+
+### Check the state
```
Comment (hikalkan) [inline docs/en/SimpleStateChecker.md]:
```suggestion
## Check the state
```

This should not be under "Definition global state checker"

File: docs/en/SimpleStateChecker.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+````csharp
+var myobj = new MyObject()
+{
+    Id = 100
+};
+
+myobj.SimpleStateCheckers.Add(new MyObjectStateChecker());
+````
+
+## Definition global state checker.
```
Comment (hikalkan) [inline docs/en/SimpleStateChecker.md]:
```suggestion
## Definition Global State Checkers
```

Please use "Title Case" for titles and do not use `.` at the end.

File: docs/en/SimpleStateChecker.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
+# Simple State Checker
+
+The **state** of some objects may depend on multiple conditions, such as some permissions are required, features need to be enabled, or it is only for special users and roles are available.
```
Comment (hikalkan) [inline docs/en/SimpleStateChecker.md]:
```suggestion
The simple state checking system can be used to enable/disable an object based on some dynamic conditions. For example, you can disable a menu item on the user interface, if the current user has not granted for a given permission. The simple state checking system provides a generic way to define and check such conditions.
```

**Review by hikalkan — CHANGES_REQUESTED** [review]

**Review by realLiangshiwei — APPROVED** [review]

**Review by hikalkan — APPROVED** [review]

**Pull request merged**